AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Models/ReleaseNotesBuilder.swift

```diff
@@ -15,24 +15,73 @@ struct ReleaseNotesBuilder {
     let repoOrg: PrepareRelease.Org
     let repoType: PrepareRelease.Repo
     let commits: [String]
+    var isTest: Bool = false
     
     // MARK: - Build
     
-    func build() -> String {
-        let contents = [
-            "## What's Changed",
-            buildCommits(),
-            .newline,
-            "**Full Changelog**: https://github.com/\(repoOrg.rawValue)/\(repoType.rawValue)/compare/\(previousVersion)...\(newVersion)"
+    func build() throws -> String {
+        let baseContent: [String] = (
+            ["## What's Changed"] +
+            buildSDKChangeSection()
+        )
+        let serviceClientChanges = repoType == .awsSdkSwift ? (try buildServiceChangeSection()) : []
+        let fullCommitLogLink = [
+            "\n**Full Changelog**: https://github.com/\(repoOrg.rawValue)/\(repoType.rawValue)/compare/\(previousVersion)...\(newVersion)"
         ]
+        let contents = baseContent + serviceClientChanges + fullCommitLogLink
         return contents.joined(separator: .newline)
     }
-    
-    // Adds a preceding `*` to each commit string
-    // This renders the list of commits as a list in markdown
-    func buildCommits() -> String {
-        commits
-            .map { "* \($0)"}
+
+    func buildSDKChangeSection() -> [String] {
+        let formattedCommits = commits
+            .filter { !$0.hasPrefix("chore") && !$0.hasPrefix("Update") }
+            .map { "* \($0)" }
+            .joined(separator: .newline)
+        if (!formattedCommits.isEmpty) {
+            return ["### AWS SDK for Swift", formattedCommits]
+        }
+        return []
+    }
+
+    func buildServiceChangeSection() throws -> [String] {
+        var features: Features
+        var mapping: [String: String]
+        // At this point, FileManager.default's current working directory is aws-sdk-swift/.
+        // The JSON files we need are located one level above it, in the workspace directory.
+        // For tests, due to sandboxing, the dummy files are created in current directory instead of
+        //  in parent directory. Therefore, fetch them from current directory for tests.
```

Comment:
Better way to do this is with dependency injection: there's a "live" object that creates the file locations for when ReleaseNotesBuilder and a "test" object that provides file locations for tests.

One or the other of these objects is passed into ReleaseNotesBuilder at creation (or, the "live" one can be used as a default param.)

These would take the place of the static methods `Features.fromFile(_:)` and `Features.mappingFromFile(_:)` below.

Reply:
Adopted suggestion; the behavior is determined by paths saved in-memory of an instance of `FeaturesReader` object that gets passed to `ReleaseNotesBuilder` initializer, with default behavior using non-test file locations.